src/App.jsx

```diff
@@ -6,20 +6,21 @@ import Appp from "./frontend/pages/product";
 import Cart from "./frontend/pages/cart";
 import Signin from "./frontend/pages/signin";
 import Signup from "./frontend/pages/signup";
+import { ProductProvider } from "./frontend/context/productListing-context";
 function App() {
   return (
     <div className="App">
       
       
-      
+      <ProductProvider>
```

Comment:
As you have already provided the product context in index.js, there is no need of providing it here again